lib/Basic/SourceLoc.cpp

```diff
@@ -100,7 +101,7 @@ SourceManager::addNewSourceBuffer(std::unique_ptr<llvm::MemoryBuffer> Buffer) {
   assert(Buffer);
   StringRef BufIdentifier = Buffer->getBufferIdentifier();
   auto ID = LLVMSourceMgr.AddNewSourceBuffer(std::move(Buffer), llvm::SMLoc());
-  BufIdentIDMap[BufIdentifier] = ID;
+  BufIdentIDMap[llvm::sys::path::convert_to_slash(BufIdentifier)] = ID;
```

Comment:
Would it make sense to only do the conversion on Windows?

Reply:
So, this is (functionally) a nop on non-Windows systems, and is explicitly documented as such. It does unnecessarily allocate a `std::string`, but I'm also not sure this is on a hot path (but maybe I'm wrong).

My thinking here was to keep this call site simple and generic, rather than do the platform switch here. But I don't feel strongly about that.

Reply:
Just to be clear, the macro buffers are only relevant for performance, right? Since `-verify-additional-file` wouldn't work for macro buffers anyways (I'd love it if it did though)

Reply:
> Just to be clear, the macro buffers are only relevant for performance, right? 
Yeah, this should be a nop for those buffers as far as I'm aware (but I have a limited understanding of how that system works).

> Since -verify-additional-file wouldn't work for macro buffers anyways (I'd love it if it did though)
That seems like it would be cool, but it's not clear to me how that would work (since macro buffers are temporary files; where would one put the `// expected-error` annotations?). Could be an interesting feature for a follow-up patch!

Reply:
Should we use `convert_to_slash` over `convert_to_native`? If the path is printed, it is better to use the native path separator as `cmd` does not like `/` over `\`.

Reply:
@compnerd I don't think this path is going to be printed. That said, I'm going to close this PR in favor of simply fixing up the tests.